test/NodeType.test.ts

```diff
@@ -7,16 +7,31 @@ import {
 } from 'graphql';
 import mockedFetch from 'node-fetch';
 
-import { HttpResource, createNodeType } from '../src';
+import { HttpResource } from '../src';
+import { config, setup } from '../src/config';
+import createResolve from '../src/types/createResolve';
+import NodeType from '../src/types/NodeType';
 import { MockContext, TestHttpApi, TestHttpResource } from './helpers';
 
 describe('NodeType', () => {
   const nodeId = Buffer.from('Widget:1').toString('base64');
 
   let schema: GraphQLSchema;
-  let context: any;
-  let NodeType;
-  let createResolve;
+  // let context: any;
+  // let NodeType;
+  // let createResolve;
+  // const { NodeType, createResolve } = rest;
+
+  const context = {
+    httpApi: new TestHttpApi(),
+  };
+
+  setup({ localIdFieldMode: 'deprecated' });
```

Comment:
this seems like a worse API, what's the reason the old one doesn't work in TS?

Reply:
the api _could actually_ be the same, but, still we'd need a static class definition. and once you do that, the old APIs are sort of a lie. The reason why we need a class def is because there are a couple of overlapping problems:

1) [TS does not infer the type of a value if the value is used in the constructor](https://www.typescriptlang.org/play/#code/MYGwhgzhAEBiD29oG8CwAoa1jwHYQBcAnAV2APiIAoAzE3YALmioEpoBeAPmjFwE92yAL4ZR6DDnwFoAS07RcAUwDucRFTaces1gG4gA). This happens when you use `X.getResource`, so you have to do `const Widget: NodeType = new NodeType`. which brings me to the second problem...
2) if a class is generated by a factory, *even if* it's a class, it's not treated as a type in TS, which means that, in order to do the above, you'd have to do weird contorsions like:
  ```ts
  const Widget: NodeTypeInterface = new NodeType()

  // the code below is not valid:
  const Widget: NodeType = new NodeType()
  // because NodeType is not a type. you could do `typeof NodeType`, but then you give up the generics
  ```

Reply:
on the bright side, having a singleton makes some things easier and avoids having to pass extra stuff around/using closures